CIP-0118/README.md

```diff
@@ -0,0 +1,466 @@
+
+### Off-Chain Component Architecture
+
+Partially valid transactions cannot be communicated across the Cardano
+network or stored in the mempool. This must be done off-chain.
+This CIP does not include an off-chain component, even though one would be necessary
+for
+
+- communicating,
+- storing, and
+- matching
+
+such transactions. The reason
+for this is that different usecases may require different off-chain architecture.
+In particular, different usecases will have specific configurations of the
+kinds of transactions or requests they are interested in engaging with.
+For example, an individual user may care only about offers of a specific token.
+An exchange may care about offers of any popular
+asset at a good price.
+
+Most importantly, off chain component does not have to be open source, nor distributed and can be used to build a for-profit business around it.
+
+
+#### Babel Service
+
+A Babel service is the catch-all name we use here to refer to a service that is interested
+in engaging with partially valid unbalanced transactions. Communication channels between such services and users constructing
+unbalanced transactions are required. Such a service can be run by anyone, probably
+alongside a full node. It would have to have a filter set for the kinds of transactions
+it will engage with. There is no guarantee that every unbalanced sub-transaction
+will ever make it on chain, since there is no guarantee that it will be eventually balanced by a subsequent incoming sub-transaction. So, each service will have
+to solve this problem for themselves, probably using an appropriate filter.
+
+#### Required Top level Guards
+
+A service processing partially valid transactions (such as the Babel service) may accept incoming transactions
+that contain non-empty `requiredTopLevelGuards`. This means this service must construct batches containing these
+transactions and also satisfy the batch-level constraints imposed by these scripts being included in the `guards` field of the top level transaction.
+
+An additional high-level language may be beneficial to specify what the guard scripts
+actually require of the batch, as Plutus script constraints may be difficult to work with directly.
+
+## Rationale: how does this CIP achieve its goals?
+
+The primary purpose of this CIP is to enable Babel Fees. Our design allows users to perform arbitrary
+swaps without two-way off-chain communication, and with counterparty irrelevance. This simulates
+fee payment in arbitrary currency whenever another use is willing
+to accept the non-primary asset offer in exchange for paying the transaction fee. This is the core
+requirement of Babel fees. In addition, the goal of this design is to lay the groundwork for
+supporting more general intents and the possibility of extending to additional ones.
+We have done so by adding the following set of features :
+
+1. transactions can be batched
```

Comment:
This "set of features" already appears more or less exactly in the _Motivation_.  I can see how it would be at home in either section: since it indicates both the feature that are needed _and_ why they've been conceived this way.  But @polinavino (cc @lehins) could you remove the duplicate in any case & ensure it's the latest content there (in case the 2 copies have diverged during editing)?

Reply:
resolved by https://github.com/cardano-foundation/CIPs/pull/862/commits/6e4e887de4199514fd141e79966f1fe23f16d288